controllers/cloudstackmachine_controller.go

```diff
@@ -179,6 +185,15 @@ func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes
 	return ctrl.Result{}, err
 }
 
+func processUserData(data []byte, r *CloudStackMachineReconciliationRunner) string {
+	// since cloudstack metadata does not allow custom data added into meta_data, following line is a hack to specify a hostname name
```

Comment:
Please avoid using words like `hack` in the source code. I suggest to use `workaround` or `trick`